lib/hammer_cli_foreman_tasks/recurring_logic.rb

```diff
@@ -26,6 +26,37 @@ class CancelCommand < HammerCLIForeman::DeleteCommand
       failure_message _('Could not cancel the recurring logic')
       build_options
     end
+    
+    class DeleteCommand < HammerCLIForeman::DeleteCommand
+      action :bulk_destroy
+
+      option '--cancelled', :flag, _("Only delete cancelled recurring logics")
+      option '--finished', :flag, _("Only delete finished recurring logics")
+      def request_params
+        params = super
+        raise ArgumentError, "Please specify if you want to remove cancelled or finished recurring logics using --cancelled or --finished " unless (options['option_cancelled'] || options["option_finished"])
+        raise ArgumentError, "Please only use one of the arguments at a time" if (options['option_cancelled'] && options["option_finished"])
+        cancelled = "state=cancelled" if options["option_cancelled"]
+        finished = "state=finished" if options["option_finished"]
+        params["search"] = cancelled || finished
+        params
+      end
+
+      command_name 'delete'
+      desc _("Delete all recuring logics filtered by the arguments")
+      success_message _('Recurring logics deleted.')
+      failure_message _('Could not delete recurring logics')
+
+      def execute
+        response = send_request
+        if response.length > 0
+          puts response.to_s
```

Comment:
With `output ListCommand.output_definition` we could use:
```suggestion
          puts _('The following recurring logics deleted:') + "\n"
          print_data(response)
```